imports/wasi_snapshot_preview1/sock.go

```diff
@@ -50,7 +51,7 @@ var sockRecv = newHostFunc(
 	"fd", "ri_data", "ri_data_len", "ri_flags", "result.ro_datalen", "result.ro_flags",
 )
 
-func sockRecvFn(_ context.Context, mod api.Module, params []uint64) syscall.Errno {
+func sockRecvFn(_ context.Context, mod api.Module, params []uint64) sys.Errno {
```

Comment:
I noticed we haven't audited POSIX, wasi-libc and wasix to figure out all the errno possibilities these should return. I'll do that afterwards. Particularly our errno should have enough values to support outbound sockets like https://github.com/stealthrocket/net

OTOH we shouldn't go too crazy on adding every definable one, as it makes a lot of work for those who aren't using posix systems. We should only add ones our apis say are returnable iotw, at least until our APIs are stable. The, as these are added, it is easier in PR review to ask "did you test this? where is this consumed in wasip1, wasix etc?" This question is harder when every errno is predefined.